src/main/java/com/app/onlinebookstore/repository/user/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.app.onlinebookstore.repository.user;
+
+import com.app.onlinebookstore.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email")
```

Comment:
I think it should work even without a query. Use the power of Spring Data

Reply:
I use this query to avoid LazyInitializationException
Is there a better approach?